contracts/sg-ics721/src/execute.rs

```diff
@@ -0,0 +1,80 @@
+use cosmwasm_std::{from_binary, to_binary, Addr, Binary, Deps, DepsMut, Env, StdResult};
+use ics721::{
+    execute::Ics721Execute,
+    state::CollectionData,
+    token_types::Class,
+    utils::{convert_owner_chain_address, get_collection_data},
+};
+use sg721_base::msg::{CollectionInfoResponse, QueryMsg};
+
+use crate::state::{SgCollectionData, SgIcs721Contract};
+
+impl Ics721Execute for SgIcs721Contract {
+    type ClassData = SgCollectionData;
+
+    /// sg-ics721 sends custom SgCollectionData, basically it extends ics721-base::state::CollectionData with additional collection_info.
+    fn get_class_data(&self, deps: &DepsMut, sender: &Addr) -> StdResult<Option<Self::ClassData>> {
+        let CollectionData {
+            owner,
+            contract_info,
+            name,
+            symbol,
+            num_tokens,
+        } = get_collection_data(deps, sender)?;
+        let collection_info: CollectionInfoResponse = deps
+            .querier
+            .query_wasm_smart(sender, &QueryMsg::CollectionInfo {})?;
+
+        Ok(Some(SgCollectionData {
+            owner,
+            contract_info,
+            name,
+            symbol,
+            num_tokens,
+            collection_info,
+        }))
+    }
+
+    fn init_msg(&self, deps: Deps, env: &Env, class: &Class) -> StdResult<Binary> {
+        let creator = match class.data.clone() {
+            None => {
+                // in case no class data is provided (e.g. due to nft-transfer module), ics721 creator is used.
+                let contract_info = deps
+                    .querier
+                    .query_wasm_contract_info(env.contract.address.to_string())?;
+                contract_info.creator
+            }
+            Some(data) => {
```

Comment:
@humanalgorithm @shanev, I changed this. Creator is now creator of ICS721 contract (and not ICS721 itself). ICS721 is instantiated with a multisig. This will be also the plan for mainnet.